src/script/descriptor.cpp

```diff
@@ -295,6 +295,8 @@ class DescriptorImpl : public Descriptor
         // Construct temporary data in `entries` and `subscripts`, to avoid producing output in case of failure.
         for (const auto& p : m_pubkey_args) {
             entries.emplace_back();
+            // If we have a cache, GetPubKey should not actually return a public key,
```

Comment:
"should not" sounds like it should be prohibited from computing the pubkey.

What about "If we have a cache, we don't need GetPubKey to compute the public key. Pass in nullptr to signify only origin info is desired.".

Reply:
Done